Fix part of #8423: Add no-unused-vars rule

Kizjkre opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes part of #8423.
2. This PR does the following: Checks for unused arguments in method signatures, but allows them to be prefixed with `unused`.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

Kizjkre committed:
Add no-unused-vars rule

Commit:
Merge remote-tracking branch 'upstream/develop' into HEAD

kevintab95 commented:
Updated the PR branch with develop to fix the following error in the CI:
```
    from proto import training_job_response_payload_pb2
ImportError: cannot import name training_job_response_payload_pb2
```
([build](https://app.circleci.com/pipelines/github/oppia/oppia/16871/workflows/0a6db576-b39b-4fa8-9c91-eb233796ad15/jobs/95192))

kevintab95 commented:
@jameesjohn do you know why the "stale" label didn't get removed automatically? I have just updated this branch with develop so the branch should not be considered "stale". Am I missing something?

jameesjohn commented:
Hi @kevintab95, I'm working on an update to automatically remove the label after the PR has been updated.
Thanks!

DubeySandeep reviewed (changes requested):
@Kizjkre, You need to fix all the lint issues raised because of the new check, can you please fix those issues?

Kizjkre committed:
Change vars ignore pattern to ignore unused vars

Kizjkre committed:
Fix all instances of unused args by prefixing with 'unused'

U8NWXD commented:
I'm reopening this at the request of @Kizjkre

Kizjkre committed:
# Conflicts:
#	core/templates/domain/question/editable-question-backend-api.service.ts
#	core/templates/domain/skill/skill-update.service.ts
#	core/templates/pages/exploration-player-page/services/exploration-recommendations.service.spec.ts
#	core/templates/pages/skill-editor-page/services/skill-editor-state.service.ts
#	extensions/interactions/ItemSelectionInput/directives/item-selection-input-validation.service.ts

Kizjkre committed:
Merge branch 'develop' into 8423-unused-args

vojtechjelinek commented:
@DubeySandeep @Kizjkre Should we allow `_` as an unused variable? Because doing `unused_` seems wrong to me, and I think we use `_` in the backend already.

Kizjkre committed:
Merge branch 'develop' into 8423-unused-args

U8NWXD commented:
@vojtechjelinek I checked with DubeySandeep, and allowing `_` is fine since we already allow that in Python

U8NWXD commented:
I'm reopening this at the request of @Kizjkre

Kizjkre committed:
Allow underscore as unused variable

Kizjkre committed:
Merge remote-tracking branch 'origin/8423-unused-args' into 8423-unused-args

Kizjkre committed:
# Conflicts:
#	core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.controller.spec.ts
#	core/tests/protractor_utils/ExplorationEditorMainTab.js

U8NWXD commented:
@DubeySandeep PTAL also

Kizjkre commented on extensions/interactions/base-validator.spec.ts:
```diff
@@ -70,7 +70,7 @@ describe('Interaction validator', function() {
   }));
 
-  beforeEach(angular.mock.inject(function($injector, $rootScope) {
+  beforeEach(angular.mock.inject(function($injector, unused$rootScope) {
```
@DubeySandeep what should I do with this? `$rootScope` is unused but the frontend tests are complaining about this.

U8NWXD replied:
Here's the frontend test failure: https://github.com/oppia/oppia/pull/11390/checks?check_run_id=1802459472#step:7:23868

vojtechjelinek commented:
@Kizjkre This PR contains many unrelated changes, you will need to fix that.

Kizjkre commented:
@vojtechjelinek I think a lot of those changes were from resolving merge conflicts with `upstream/develop`.

vojtechjelinek commented:
@Kizjkre There is some weird error with how GitHub shows this PR, could you please try merging from develop again?

sajalasati commented:
@Kizjkre Please try to fix the merge conflicts in this PR and push the changes again via new commit(s). You can do so by following the instructions mentioned in our wiki [guide](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#address-review-comments-until-all-reviewers-give-lgtm-looks-good-to-me) on this. Let us know if you're facing any difficulties. Thanks!

Kizjkre commented:
Go to #11874.
